### DISCUSSION TITLE
Does Biome follow semantic versioning and focus on stability?

### DISCUSSION THREAD

**Discussion by anonrig** [biomejs/biome#1466] [General]
The latest release v1.5.0 broke the formatter for Sentry JavaScript repository, and it gives 195 errors more with the same exact configuration. This makes me realize how Biome doesn't follow semantic versioning principles.

I'd like to ask for the official opinion for the Biome core team about this, since for more enterprise and stable repositories, these breaking changes are a no-go for semver-minor releases.

As an example, here's the lint result of Biome after moving to v1.5.0

```
❯ yarn lint:biome
yarn run v1.22.19
$ yarn biome check .
$ /Users/yagiz/coding/sentry-javascript/node_modules/.bin/biome check .
./nx.json lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file ./nx.json was ignored.


./yarn.lock project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./yarn.lock is protected because is handled by another tool. Biome won't process it.


./.gitignore project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./.gitignore is protected because is handled by another tool. Biome won't process it.


./package.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./package.json is protected because is handled by another tool. Biome won't process it.


./tsconfig.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./tsconfig.json is protected because is handled by another tool. Biome won't process it.


./tsconfig.dev.json lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file ./tsconfig.dev.json was ignored.


./tsconfig.dev.json check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file contains diagnostics that needs to be addressed.


./nx.json check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file contains diagnostics that needs to be addressed.


./lerna.json lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file ./lerna.json was ignored.


./lerna.json check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file contains diagnostics that needs to be addressed.


./dev-packages/overhead-metrics/.gitignore project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./dev-packages/overhead-metrics/.gitignore is protected because is handled by another tool. Biome won't process it.


./dev-packages/overhead-metrics/package.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./dev-packages/overhead-metrics/package.json is protected because is handled by another tool. Biome won't process it.


./packages/deno/tsconfig.types.json lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file ./packages/deno/tsconfig.types.json was ignored.


./dev-packages/overhead-metrics/tsconfig.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./dev-packages/overhead-metrics/tsconfig.json is protected because is handled by another tool. Biome won't process it.


./packages/deno/package.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./packages/deno/package.json is protected because is handled by another tool. Biome won't process it.


./packages/deno/tsconfig.types.json check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file contains diagnostics that needs to be addressed.


./packages/deno/tsconfig.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./packages/deno/tsconfig.json is protected because is handled by another tool. Biome won't process it.


./dev-packages/rollup-utils/package.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./dev-packages/rollup-utils/package.json is protected because is handled by another tool. Biome won't process it.


./packages/tracing/package.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./packages/tracing/package.json is protected because is handled by another tool. Biome won't process it.


./packages/tracing/tsconfig.json project ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The file ./packages/tracing/tsconfig.json is protected because is handled by another tool. Biome won't process it.


The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 460.
Checked 2412 file(s) in 173ms
Found 195 error(s)
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

**Comment by ematipico** [comment]
@anonrig We have a page where we explain **our** semantic versioning, which is the contract that **we try** to follow:  https://biomejs.dev/internals/versioning/

**Reply by anonrig** [reply]
I couldn't find a reference to those changes such as requiring `**` prefix in a minor release. I think the perception of these `custom` semantic versioning rules, would be a deal-breaker for most of the companies. And on top of that, these breaking changes were not included in the release blog post.

For example, in previous release, ignored files were not giving an error. But right now:

```
./packages/tracing/tsconfig.types.json lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file ./packages/tracing/tsconfig.types.json was ignored.
```

**Reply by ematipico** [reply]
There are different things to unpack here; I am going to break them down:

- Blog posts are a **marketing** tool; we use them to highlight the features we deem remarkable and show off. They were never meant to be copies of our changes from one release to another. In fact, we don't aim to release a blog post for each release. That's why we have the changelogs, **which we try to keep up to date** because we still have a manual process for those. If you are interested in detailed changes, I would suggest you check the [changelog](https://biomejs.dev/internals/changelog/) and/or the history of the PRs.
- > For example, in previous release, ignored files were not giving an error:
  
  Have you considered that this was a bug? If you haven't found anything in the changelog related to this, that's definitely an **unintentional** change. I wouldn't jump to conclusions about breaking changes and such.

- > it gives 195 errors more with the same exact configuration. This makes me realize how Biome doesn't follow semantic versioning principles

	That's how *our contract* of [semantic versioning works](https://biomejs.dev/internals/versioning/), and we *try* to adhere to it as much as possible by striking a balance between innovation and developer experience. Still, Biome is a developer tool, and that's why we tell users **to NOT install** Biome using ranges because even a bug fix can block your CI e.g. we fix a false positive inside a lint rule, and now you'll get a new diagnostic. As the tool gets better, it's able to catch more stuff for you, hence you'll have to fix the code.
	
	This doesn't mean that all new errors are unintentional, as I said before.

**Comment by Conaclos** [comment]
We are open to any suggestions to refine our versioning philosophy in order to bring more control and stability to our users.
Still, we recommend using exact version (`npm i --save-exact`). This is also what Prettier recommends.

**Comment by Yovach** [comment]
Why does the message `The file [filename] was ignored.` is throwing an error? It was ignored, fine, but I don't want it to crash my program :/

**Reply by Conaclos** [reply]
I think you are getting the error reported in #1471

Could you add a comment in the issue and show your config?

**Reply by Yovach** [reply]
I commented the issue with my config, thanks
